ShowerShow/Control/ShowerThoughtController.cs

```diff
@@ -0,0 +1,196 @@
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Azure.Functions.Worker;
+using Microsoft.Azure.Functions.Worker.Http;
+using Microsoft.Azure.WebJobs.Extensions.OpenApi.Core.Attributes;
+using Microsoft.Extensions.Logging;
+using Microsoft.OpenApi.Models;
+using Newtonsoft.Json;
+using System.IO;
+using System.Net;
+using System.Threading.Tasks;
+using System;
+using ShowerShow.Models;
+using ShowerShow.DTO;
+using ShowerShow.Repository.Interface;
+using System.Collections.Generic;
+using System.Collections;
+using ShowerShow.Model;
+using ShowerShow.Service;
+using System.Configuration;
+
+namespace ShowerShow.Controllers
+{
+    public class ShowerThoughtController
+    {
+        private readonly ILogger<ShowerThoughtController> _logger;
+        private IShowerThoughtService showerThoughtService;
+        private IUserService userService;
+
+        public ShowerThoughtController(ILogger<ShowerThoughtController> log, IShowerThoughtService showerThoughtService, IUserService userService)
+        {
+            _logger = log;
+            this.showerThoughtService = showerThoughtService;
+            this.userService = userService;
+        }
+        [Function("CreateShowerThought")]
+        [OpenApiOperation(operationId: "CreateShowerThought", tags: new[] { "ShowerThoughts" })]
+        [OpenApiRequestBody("application/json", typeof(ShowerThoughtDTO), Description = "The shower thought data.")]
+        [OpenApiParameter(name: "ShowerId", In = ParameterLocation.Path, Required = true, Type = typeof(Guid), Description = "The ShowerId parameter")]
+        [OpenApiParameter(name: "UserId", In = ParameterLocation.Query, Required = false, Type = typeof(Guid), Description = "The User ID parameter")]
+        [OpenApiResponseWithBody(statusCode: HttpStatusCode.Created, contentType: "application/json", bodyType: typeof(ShowerThought), Description = "The OK response with the new thought.")]
+        public async Task<HttpResponseData> CreateShowerThought([HttpTrigger(AuthorizationLevel.Anonymous, "post", Route = "shower/thoughts/{ShowerId:Guid}")] HttpRequestData req, Guid ShowerId, Guid UserId)
+        {
+            _logger.LogInformation("Creating new thought.");
+            string requestBody = await new StreamReader(req.Body).ReadToEndAsync();
+            HttpResponseData responseData = req.CreateResponse();
+
+            if (await userService.CheckIfUserExist(UserId))
```

Comment:
There is a method called There is another method called check if user exist and active. Can you use that method instead for showethought endpoints?